### ISSUE TITLE
Accessibility: Add aria attributes to site tree

### ISSUE THREAD

**Issue by emteknetnz** [silverstripe/silverstripe-admin#2000]
Split from - https://github.com/silverstripe/silverstripe-cms/issues/3101

[WAI-ARIA Tree View Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/treeview/)

The CMS page site tree lacks aria attributes

## Context

The site tree in `/admin/pages/` had almost no keyboard interactivity even though there were various functions possible with mouse interaction.

The relevant keyboard interaction will have been added through the "related" issues below, but screen readers probably don't give necessary hints to blind users.

### Main Container
- The main container should have `role="tree"`. This tells screen readers that the element contains a hierarchical structure.

### Pages
- Each page item within the tree should have `role="treeitem"`.
- To indicate nesting levels, use `aria-level` (e.g., `aria-level="1"` for top-level pages, `aria-level="2"` for their children).
- For pages with nested children, use `aria-expanded="false"` when the list is collapsed and `aria-expanded="true"` when it's open.

### Expand/Collapse Arrow
- Icons controlling the expand/collapse action should have a dynamic and descriptive `aria-label`, such as "Toggle child pages".

### Mouse Drag-and-Drop
- ~The page being dragged should have `aria-grabbed="true"` while in motion.~
- ~The invisible drop zones where the page can be inserted should have `aria-dropeffect="move"`.~
- [aria-grabbed](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-grabbed) and [aria-dropeffect](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-dropeffect) are both deprecated in ARIA 1.1 and instead the current recommendation is to provide a non-dragging alternative i.e. "keyboard move mode" - https://www.w3.org/WAI/WCAG22/Understanding/dragging-movements.html

### Notes
- The `<ins>` elements with `tabindex="0"` added in https://github.com/silverstripe/silverstripe-cms/pull/3116 will need an appropriate role added

## Related
- https://github.com/silverstripe/silverstripe-admin/issues/1957
- https://github.com/silverstripe/silverstripe-cms/issues/3101
- https://github.com/silverstripe/silverstripe-cms/issues/3115

## Acceptance criteria
- [ ] Screen readers give sensible instructions and labels where appropriate
  - [ ] Whether nested tree is expanded or not, and how to expand/collapse it
  - [ ] Instructions for updating the sort order (if that hasn't been implemented yet, move this part to a separate card and block it on https://github.com/silverstripe/silverstripe-cms/issues/3115)
- [ ] Icons related to the above actions are not read out (i.e. linked issue above)

## Kitchen-sink CI
- https://github.com/creative-commoners/recipe-kitchen-sink/actions/runs/23723682711
^ behat failure is [existing](https://github.com/silverstripe/recipe-kitchen-sink/actions/runs/23722180785/job/69098964952#step:12:476)

## PRs
- https://github.com/silverstripe/silverstripe-framework/pull/11863
- https://github.com/silverstripe/silverstripe-cms/pull/3120
- https://github.com/silverstripe/silverstripe-admin/pull/2016
- https://github.com/silverstripe/developer-docs/pull/827

**Comment by GuySartorelli** [comment]
The `ins` element that was added to the tab index in https://github.com/silverstripe/silverstripe-cms/pull/3116 needs an appropriate `role` or perhaps should be replaced with a semantic `button` HTML element.

**Comment by emteknetnz** [comment]
@GuySartorelli I'm going leave these PRs open for now and block this issue pending https://github.com/silverstripeltd/product-issues/issues/919 (private issue, to do with working out a better system for keyboard nav) - I'm not really happy with how this issue is progressing and I have a feeling we'll just end up changing how the keyboard nav works on the site tree in general (it's currently wrong, i.e., space probably shouldn't toggle showing nested pages), keyboard move functionality doesn't work, we may not even retain tabbing through pages and instead use arrows keys only. I feel we need to get all of that worked out first before checking the screen reader experience is correct
